### PR TITLE
Basic functions

### PR DESCRIPTION
**Pull request by nicolbidone**
Screenshot_1553263830

### CODE REVIEW

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt [hunk 1]
```diff
@@ -2,33 +2,62 @@ package com.example.nicocalculator
...
+import com.example.nicocalculator.mvp.presenter.Presenter
+import com.example.nicocalculator.mvp.view.KeyView
 import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
 
-
     companion object {
         const val OPERATIONS_DEFAULT_TEXT = "Operations"
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: pass this constants to the model pls

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt [hunk 2]
```diff
@@ -2,33 +2,62 @@ package com.example.nicocalculator
...
+
+        for (button in buttons) {
+            button.setOnClickListener {
+                presenter.onKeyButtonPressed(button)
+            }
+        }
     }
 }
+
+/*
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/MainActivity.kt]: If you will not use this code please remove it

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt [hunk 3]
```diff
@@ -0,0 +1,117 @@
+package com.example.nicocalculator.mvp.presenter
+
+import android.widget.Button
+import com.example.nicocalculator.Key
+import com.example.nicocalculator.MainActivity
+import com.example.nicocalculator.mvp.model.CalculatorModel
+import com.example.nicocalculator.mvp.view.KeyView
+
+class Presenter(private val model: CalculatorModel, private val view: KeyView) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: Please try to use the same name for the presenter, model and view. In this case, I think that can be CalculatorPresenter, CalculatorModel and CalculatorView

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        var textResults = VALUE_EMPTY_TEXT
+        var valeant = mutableListOf<Key>()
+
+        for (vale in itemList) {
+            textResults += "$vale "
+            if (valeant.size < 2) valeant.add(Key(vale.getValue()))
+            else {
+                var aux = VALUE_ZERO
+                when (valeant[1].getValue()) {
+                    "+" -> aux = valeant[0].getInt() + vale.getInt()
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: put the operators in constants pls

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+                num = VALUE_EMPTY_TEXT
+                res.add(Key(car.toString()))
+            } else num += car.toString()
+        }
+        if (num != VALUE_EMPTY_TEXT) res.add(Key(num))
+        return res
+    }
+}
+
+/*fun unregister() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: please remove all the commented code pls

File: app/src/main/java/com/example/nicocalculator/mvp/view/KeyView.kt [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+        activity.text_results.text = string
+    }
+
+    fun setTextOperations(string: String) {
+        activity.text_operations.text = string
+    }
+
+}
+
+/*fun deleteButtonPressed() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/nicocalculator/mvp/view/KeyView.kt]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 1 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/java/com/example/nicocalculator/MainActivity.kt (hunk 2 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/MainActivity.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt (hunk 5 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/view/KeyView.kt (hunk 6 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/view/KeyView.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt (hunk 4 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt (hunk 3 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+        view.setTextOperations(textOperations)
+    }
+
+    fun onEqualButtonPressed() {
+        var textResult = getResults(getItems(model.getTextOperations()))
+        model.setTextResult(textResult)
+        view.setTextResult(textResult)
+    }
+
+    private fun getResults(itemList: List<Key>): String {
```
Comment (priettt) [inline app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt]: You could move this to the model: if in the future a change is made in the way the results are handled, the presenter will  remain the same, ignoring that change, and you will only need to modify the model.

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                }
+                textResults += " = $aux\n$aux "
+                valeant = mutableListOf()
+                valeant.add(Key(aux.toString()))
+            }
+        }
+        return textResults
+    }
+
+    private fun getItems(items: CharSequence): List<Key> {
```
Comment (priettt) [inline app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt]: This could be moved to the model, for the same reason.

Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt]: Done

File: app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt (hunk 7 above)
Comment (nicolbidone) [reply app/src/main/java/com/example/nicocalculator/mvp/presenter/CalculatorPresenter.kt]: Done

**Review by SartiMau — APPROVED** [review]

**Review by capraber — APPROVED** [review]

**Review by priettt — APPROVED** [review]
For your next PR, make the CalculatorView extend a view with a weak reference, as in the example.

**Pull request merged**